tensorflow/c/tf_tensor.h

```diff
@@ -45,6 +46,17 @@ limitations under the License.
 extern "C" {
 #endif
 
+// Allocator Attributes used for tensor allocation. 
+typedef struct TF_AllocatorAttributes { 
+  size_t struct_size; 
+  // Set boolean to 1 for CPU allocation, else 0. 
+  unsigned char on_host; 
```

Comment:
Comment from @yisitu: Perhaps bring TF_Bool  into c_api_macros as well?

What Yi Situ means is basically adding this to c_api_macros.h:
```cpp
#ifndef TF_BOOL_DEFINED
#define TF_BOOL unsigned char
#endif // TF_BOOL_DEFINED
```
Then, you can use `TF_BOOL` here instead of `unsigned char`.